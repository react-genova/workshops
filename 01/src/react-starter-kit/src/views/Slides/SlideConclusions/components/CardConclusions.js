//@flow
import React from 'react';
import { Card, CardElement } from '../../../Card';
import LabelTerminal from '../../../LabelTerminal';

const CardConclusions = () => (
    <Card title="Step by step installation guide" description="All copy and no paste makes Jack a dull boy">
        <CardElement title="Just the first time" description="" url="" >
            <CardElement title="Download and install Node.js" description="" url="" />
            <CardElement title="Download and install Visual Studio Code and its plugins" description="" url="" />
            <LabelTerminal text="npm install flow-bin flow-typed" />
            <LabelTerminal text="npm install create-react-app" />
        </CardElement>
        <CardElement title="Each time you create an app" description="" url="" >
            <LabelTerminal text="create-react-app your-app-pretty-cool-name" />
            <LabelTerminal text="cd your-app-pretty-cool-name" />
            <LabelTerminal text="npm install --save styled-components recompose react-redux flow-bin redux redux-immutable redux-logger reselect axios axios-mock-adapter immutable lodash enzyme jest-immutable-matchers enzyme-adapter-react-16 enzyme-to-json react-styleguidist react-test-renderer redux-test-utils" />
            <LabelTerminal text="flow-typed install enzyme@3.0.0 jest@20.0.0 lodash@4.0.0 recompose@0.24.0 styled-components@2.2.3 reselect@3.0.1 axios@0.17.1" />
            <CardElement title="Configure your 'package.json' file" description="" url="https://github.com/react-genova/workshops/blob/master/01/resources/pacakge.json.addons" />
            <CardElement title="Add linter config file '.eslintrc'" description="" url="https://github.com/react-genova/workshops/blob/master/01/resources/.eslintrc" />
            <CardElement title="Add flow config file '.flowconfig'" description="" url="https://github.com/react-genova/workshops/blob/master/01/resources/.flowconfig" />
            <CardElement title="Add styleguidist config file 'styleguide.config.js'" description="" url="https://github.com/react-genova/workshops/blob/master/01/resources/styleguide.config.js" />
            <CardElement title="Add test config file 'setupTests.js'" description="" url="https://github.com/react-genova/workshops/blob/master/01/resources/setupTests.js" />
        </CardElement>
    </Card>
);

export default CardConclusions;