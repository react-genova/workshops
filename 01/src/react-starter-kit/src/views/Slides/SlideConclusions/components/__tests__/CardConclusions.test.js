//@flow
import React from 'react';
import { shallow } from 'enzyme';
import CardConclusions from '../CardConclusions';

describe('Card Conclusions', () => {

    it('snapshot check', () => {
        const component = shallow(<CardConclusions />);
        expect(component).toMatchSnapshot();
    });

});