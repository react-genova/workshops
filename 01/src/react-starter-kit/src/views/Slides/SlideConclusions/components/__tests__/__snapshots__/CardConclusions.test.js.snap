// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Conclusions snapshot check 1`] = `
<Card
  description="All copy and no paste makes Jack a dull boy"
  title="Step by step installation guide"
>
  <CardElement
    description=""
    title="Just the first time"
    url=""
  >
    <CardElement
      description=""
      title="Download and install Node.js"
      url=""
    />
    <CardElement
      description=""
      title="Download and install Visual Studio Code and its plugins"
      url=""
    />
    <LabelTerminal
      text="npm install flow-bin flow-typed"
    />
  </CardElement>
  <CardElement
    description=""
    title="Each time you create an app"
    url=""
  >
    <LabelTerminal
      text="npm install create-react-app"
    />
    <LabelTerminal
      text="create-react-app your-app-pretty-cool-name"
    />
    <LabelTerminal
      text="cd your-app-pretty-cool-name"
    />
    <LabelTerminal
      text="npm install --save styled-components recompose react-redux flow-bin redux redux-immutable redux-logger reselect axios axios-mock-adapter immutable lodash enzyme jest-immutable-matchers enzyme-adapter-react-16 enzyme-to-json react-styleguidist react-test-renderer redux-test-utils"
    />
    <LabelTerminal
      text="flow-typed install enzyme@3.0.0 jest@20.0.0 lodash@4.0.0 recompose@0.24.0 styled-components@2.2.3 reselect@3.0.1 axios@0.17.1"
    />
    <CardElement
      description=""
      title="Configure your 'package.json' file"
      url="https://github.com/react-genova/workshops/blob/master/01/resources/pacakge.json.addons"
    />
    <CardElement
      description=""
      title="Add linter config file '.eslintrc'"
      url="https://github.com/react-genova/workshops/blob/master/01/resources/.eslintrc"
    />
    <CardElement
      description=""
      title="Add flow config file '.flowconfig'"
      url="https://github.com/react-genova/workshops/blob/master/01/resources/.flowconfig"
    />
    <CardElement
      description=""
      title="Add styleguidist config file 'styleguide.config.js'"
      url="https://github.com/react-genova/workshops/blob/master/01/resources/styleguide.config.js"
    />
    <CardElement
      description=""
      title="Add test config file 'setupTests.js'"
      url="https://github.com/react-genova/workshops/blob/master/01/resources/setupTests.js"
    />
  </CardElement>
</Card>
`;
